visualization/app/codeCharta/state/store/appSettings/amountOfTopLabels/amountOfTopLabels.selector.ts

```diff
@@ -1,4 +1,23 @@
 import { createSelector } from "../../../angular-redux/createSelector"
 import { appSettingsSelector } from "../appSettings.selector"
+import { AppSettings } from "../../../../codeCharta.model"
 
-export const amountOfTopLabelsSelector = createSelector([appSettingsSelector], appSettings => appSettings.amountOfTopLabels)
+import { numberOfFilesSelector } from "./numberOfFiles.selector"
+
+const NUMBER_OF_BUILDINGS = 100
+const DEFAULT_NUMBER_OF_LABELS = 1
+const MAX_NUMBER_OF_LABELS = 10
+
+export const getAmountOfTopLabels = (appSettings: Pick<AppSettings, "amountOfTopLabels">, numberOfFiles: any): number => {
+	const { amountOfTopLabels } = appSettings
+	const numberOfLabels = Math.floor(numberOfFiles.length / NUMBER_OF_BUILDINGS)
+	if (numberOfLabels <= DEFAULT_NUMBER_OF_LABELS) {
+		return amountOfTopLabels
+	}
+	if (numberOfLabels > DEFAULT_NUMBER_OF_LABELS && numberOfLabels <= MAX_NUMBER_OF_LABELS) {
+		return numberOfLabels
+	}
+	return MAX_NUMBER_OF_LABELS
```

Comment:
```suggestion
	const numberOfLabels = Math.floor(numberOfFiles.length / NUMBER_OF_BUILDINGS)
	
	return Math.min(numberOfLabels, amountOfTopLabels, MAX_NUMBER_OF_LABELS)
```

Reply:
I thought you want to see 1 Label per 100 buildings but a maximum of 10 labels, right? With this suggested function, I don't get this result.

Reply:
This is used as default, right? `amountOfTopLabels` is probably too much in that case as it's not yet defined by the user. If it's defined, that would overrule anything else.